test/spec.journey-model.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const JourneyModel = require('../lib/journey-model');
+const SessionModel = require('../lib/model');
+
+describe('session model', () => {
+
+    it('exports a function', () => {
+        SessionModel.should.be.a.function;
+    });
+
+    it('should be an instance of SessionModel', () => {
+        let journeyModel = new JourneyModel(null, { key: 'test', session: {} });
+        journeyModel.should.be.an.instanceOf(SessionModel);
+    });
+
+    describe('registerModel', () => {
+        it('should add the model to the list of currently registered models', () => {
+            let session = {
+                test: {}
+            };
+            let modelStub = {
+                options: { key: 'test1' }
+            };
+            let journeyModel = new JourneyModel(null, { key: 'test', session });
+            journeyModel.registerModel(modelStub);
+            journeyModel.registeredModels.should.deep.equal([
+                modelStub
+            ]);
+        });
+
+        it('should add the model to the list of previously registered models', () => {
+            let session = {
+                test: {}
+            };
+            let modelStub = {
+                options: { key: 'test1' }
+            };
+            let journeyModel = new JourneyModel(null, { key: 'test', session });
+            journeyModel.registerModel(modelStub);
+            session.test['registered-models'].should.deep.equal(['test1']);
+        });
+
+        it('should not add duplicate model keys to the model list', () => {
+            let session = {
+                test: {
+                    'registered-models': [ 'test1' ]
+                }
+            };
+            let modelStub = {
+                options: { key: 'test1' }
+            };
+            let journeyModel = new JourneyModel(null, { key: 'test', session });
```

Comment:
Is there any way to make this test clearer? It took a little bit of thinking to separate out the `test1` key lines from the `test` key line.